### ISSUE TITLE
convert onnx to trt failed(invalid node convtranspose)

### ISSUE THREAD

**Issue by hererookie** [NVIDIA/TensorRT#1654]
## Description

I try to convert the onnx to trt, the onnx convert trt successful and work fine(on tensorrt Version 7.1.3);but failed on tensorrt 8.0.1.6-1;
I use:python -m tf2onnx.convert --saved-model=save_model/nonms/ --opset=11 --output=model.onnx --tag=serve 
to output the onnx model;

[12/01/2021-19:44:56] [V] [TRT] Running deconvolution with: 
Padding mode: NOTSET
Pre-padding: (0, 0)
Post-padding: (0, 0)
[12/01/2021-19:44:56] [V] [TRT] Registering layer: ConvTranspose__377 for ONNX node: ConvTranspose__377
[12/01/2021-19:44:56] [E] Error[1]: [network.cpp::setWeightsName::3014] Error Code 1: Internal Error (The given weights is not used in the network!)
[12/01/2021-19:44:56] [V] [TRT] Padding output deconvolution tensor with: (1, 1)
[12/01/2021-19:44:56] [E] [TRT] ModelImporter.cpp:720: While parsing node number 56 [ConvTranspose -> "ConvTranspose__377:0"]:
[12/01/2021-19:44:56] [E] [TRT] ModelImporter.cpp:721: --- Begin node ---
[12/01/2021-19:44:56] [E] [TRT] ModelImporter.cpp:722: input: "StatefulPartitionedCall/functional_3/leaky_re_lu_17/LeakyRelu:0"
input: "StatefulPartitionedCall/functional_3/conv2d_transpose/conv2d_transpose/ReadVariableOp:0"
input: "StatefulPartitionedCall/functional_3/conv2d_transpose/BiasAdd/ReadVariableOp:0"
output: "ConvTranspose__377:0"
name: "ConvTranspose__377"
op_type: "ConvTranspose"
attribute {
  name: "dilations"
  ints: 1
  ints: 1
  type: INTS
}
attribute {
  name: "strides"
  ints: 2
  ints: 2
  type: INTS
}
attribute {
  name: "kernel_shape"
  ints: 1
  ints: 1
  type: INTS
}
attribute {
  name: "output_shape"
  ints: 26
  ints: 26
  type: INTS
}

[12/01/2021-19:44:56] [E] [TRT] ModelImporter.cpp:723: --- End node ---
[12/01/2021-19:44:56] [E] [TRT] ModelImporter.cpp:726: ERROR: ModelImporter.cpp:166 In function parseGraph:
[6] Invalid Node - ConvTranspose__377
[network.cpp::setWeightsName::3014] Error Code 1: Internal Error (The given weights is not used in the network!)
[12/01/2021-19:44:56] [E] Failed to parse onnx file
[12/01/2021-19:44:56] [I] Finish parsing network model
[12/01/2021-19:44:56] [E] Parsing model failed
[12/01/2021-19:44:56] [E] Engine creation failed
[12/01/2021-19:44:56] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8001] # /usr/src/tensorrt/bin/trtexec --onnx=model_2_416_416_3_trt_nms.onnx --saveEngine=model_BatchedNMS_TRT.engine --verbose

## Environment

**TensorRT Version**: 8.0.1.6-1
**ONNX-TensorRT Version / Branch**:
**GPU Type**: 
**Nvidia Driver Version**: 
**CUDA Version**: cuda10.2
**CUDNN Version**: 
**Operating System + Version**: jetson
**Python Version (if applicable)**: 
**TensorFlow + TF2ONNX Version (if applicable)**: tf2.3, TF2ONNX:1.8.5
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**:

**Comment by zerollzeng** [comment]
Hi @hererookie, Can you share the onnx model?

**Comment by hererookie** [comment]
[daerwen_nonms_0.onnx.zip](https://github.com/NVIDIA/TensorRT/files/7638611/daerwen_nonms_0.onnx.zip)
@zerollzeng

**Comment by hererookie** [comment]
> Hi @hererookie, Can you share the onnx model?

hi my  jetpack version is 4.6

**Comment by hererookie** [comment]
> Hi @hererookie, Can you share the onnx model?

@zerollzeng hanks for your reply, the onnx file is
[daerwen_nonms_0.onnx.zip](https://github.com/NVIDIA/TensorRT/files/7639301/daerwen_nonms_0.onnx.zip)

**Comment by zerollzeng** [comment]
This has been fixed in TRT 8.2, also refer to https://github.com/NVIDIA/TensorRT/issues/1587

**Comment by hererookie** [comment]
> This has been fixed in TRT 8.2, also refer to #1587

refer to #1587 ,that mean aren't supported 1D deconvolutions; but on  the tensorrt8 operator support matrix explicitly mentions [https://github.com/onnx/onnx-tensorrt/blob/8.0-EA/docs/operators.md](url) that support convtranspose 2D and 3D;not support convtranspose 1D; In addition , in my onnx model only use convtranspose 2D; so TensorRT 8.0 still not support 2D deconvolutions ??? and Jetson not apply tensorrt8.2

**Comment by hererookie** [comment]
> This has been fixed in TRT 8.2, also refer to #1587

@zerollzeng hi refer to #1587 ,that mean aren't supported 1D deconvolutions; but on the tensorrt8 operator support matrix explicitly mentions https://github.com/onnx/onnx-tensorrt/blob/8.0-EA/docs/operators.md that support convtranspose 2D and 3D;not support convtranspose 1D; In addition , in my onnx model use convtranspose 2D only ; so TensorRT 8.0 still not support 2D deconvolutions ? Another sad problem is that tensorrt8.2 cannot be applied on Jetson.

**Comment by hererookie** [comment]
> This has been fixed in TRT 8.2, also refer to #1587

@zerollzeng hi ,if tensorrt8.0 not support  convtranspose 2D, Do you know when we can use tensorrt8.2 in jetpack? New jetpack version include tesnorrt8.2 ? thanks

**Comment by zerollzeng** [comment]
Actually, I just retest your model with the desktop 8.2 version, but trt 8.0 should support 2d convtranspose.  I guess it's a bug in 8.0 and has been fixed in 8.2. 

I don't know the date with the new jetpack release with trt 8.2, this is my personal Github account so I don't represent Nvidia here. just notice your issue and try to provide some help :)

**Comment by zerollzeng** [comment]
Perhaps you can fix it by modifying the onnx, I will try to do some investigation if I have spare time :)

**Comment by hererookie** [comment]
> Perhaps you can fix it by modifying the onnx, I will try to do some investigation if I have spare time :)
thank you for reply; modify the onnx mean replace the convtranspose with upsample?

**Comment by hererookie** [comment]
> Actually, I just retest your model with the desktop 8.2 version, but trt 8.0 should support 2d convtranspose. I guess it's a bug in 8.0 and has been fixed in 8.2.
> 
> I don't know the date with the new jetpack release with trt 8.2, this is my personal Github account so I don't represent Nvidia here. just notice your issue and try to provide some help :)

@

> Actually, I just retest your model with the desktop 8.2 version, but trt 8.0 should support 2d convtranspose. I guess it's a bug in 8.0 and has been fixed in 8.2.
> 
> I don't know the date with the new jetpack release with trt 8.2, this is my personal Github account so I don't represent Nvidia here. just notice your issue and try to provide some help :)

thank you for reply; modify the onnx mean replace the convtranspose with upsample?

**Comment by hererookie** [comment]
> Perhaps you can fix it by modifying the onnx, I will try to do some investigation if I have spare time :)

thank you for reply; modify the onnx mean replace the convtranspose with upsample?

**Comment by hererookie** [comment]
> We have fix some onnx parser issue for `ConvTranspose` in rel-8.2, another try worth doing is to cross compile onnx parser,
> 
> maybe replace this function: https://github.com/onnx/onnx-tensorrt/blob/release/8.2-GA/builtin_op_importers.cpp#L726

hi、thanks for your advice; you mean convert the onnx to trt on tensorrt8.2,then run it on trt 8.0?

**Comment by ttyio** [comment]
Hello @hererookie , the 8.2 plan file cannot run in trt 8.0 enviroment. My suggestion is to manually build the 8.0 parser with the `ConvTranspose` fix in 8.2, the steps could be:

let's checkout `release/8.0` branch source code, and patch the `ConvTranspose` function using the `release/8.2` implementation in https://github.com/onnx/onnx-tensorrt/blob/release/8.2-GA/builtin_op_importers.cpp#L726

then follow the cross compile instruction in https://github.com/NVIDIA/TensorRT/tree/release/8.0#building-tensorrt-oss, built the onnx2trt parser, and replace the parser in your system.

**Comment by hererookie** [comment]
> Hello @hererookie , the 8.2 plan file cannot run in trt 8.0 enviroment. My suggestion is to manually build the 8.0 parser with the `ConvTranspose` fix in 8.2, the steps could be:
> 
> let's checkout `release/8.0` branch source code https://github.com/NVIDIA/TensorRT/tree/release/8.0, and patch the `ConvTranspose` function using the `release/8.2` implementation in https://github.com/onnx/onnx-tensorrt/blob/release/8.2-GA/builtin_op_importers.cpp#L726
> 
> then follow the cross compile instruction in https://github.com/NVIDIA/TensorRT/tree/release/8.0#building-tensorrt-oss, built the onnx2trt parser, and replace the parser in your system.

OK、i will try it，thank you very much

**Comment by ttyio** [comment]
@hererookie cool, thank you too!

**Comment by ysyyork** [comment]
Is there any timeline to integrate this into jetson jetpack? Or should I ask this question in a different place?

**Comment by ttyio** [comment]
@ysyyork , maybe ask in https://forums.developer.nvidia.com/? the Jetpack release schedule is out of TRT's control, thanks!

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, thanks!
